### PR TITLE
WIP - update to django 1.10

### PR DESCRIPTION
**Pull request by luzfcb**
changes:
- [x] rename MIDDLEWARE_CLASSES to MIDDLEWARE
- [x] add default AUTH_PASSWORD_VALIDATORS generated by django 1.10 startproject
- [x] update `manage.py` to use same code of `manage.py` from django 1.10
- [x] <s>add LocaleMiddleware to fix issue #792</s> (It will be included in a separated pull-request)
- [x] update django version on `setup.py`
- [x] make tests pass by passing new password auth rules
- [x] update python dependencies
- [x] update to bootstrap 4 alfa 4, jquery and tether.js. (i generate the integrity hash using https://www.srihash.org/ . this was indicated by documentation https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity
- [x] sync `sites` app migrations with django 1.10, and fix aditional migrations to `sites` and `user` app

- [ ] please, not merge, still requires checks everything that was said in https://github.com/pydanny/cookiecutter-django/issues/703

### CODE REVIEW

File: {{cookiecutter.project_slug}}/config/settings/common.py [hunk 1]
```diff
@@ -53,9 +53,10 @@
 
 # MIDDLEWARE CONFIGURATION
 # ------------------------------------------------------------------------------
-MIDDLEWARE_CLASSES = (
+MIDDLEWARE = (
     'django.middleware.security.SecurityMiddleware',
     'django.contrib.sessions.middleware.SessionMiddleware',
+    'django.middleware.locale.LocaleMiddleware',
```
Comment (pydanny) [inline {{cookiecutter.project_slug}}/config/settings/common.py]: I know we should include it, but is `LocaleMiddleware` a requirement for Django 1.10?

File: {{cookiecutter.project_slug}}/config/settings/local.py [hunk 2]
```diff
@@ -50,7 +50,7 @@
 
 # django-debug-toolbar
 # ------------------------------------------------------------------------------
-MIDDLEWARE_CLASSES += ('debug_toolbar.middleware.DebugToolbarMiddleware',)
+MIDDLEWARE += ('debug_toolbar.middleware.DebugToolbarMiddleware',)
```
Comment (pydanny) [inline {{cookiecutter.project_slug}}/config/settings/local.py]: Sigh. I know this is necessary, but I'm not the biggest fan of the middleware changes. 😩

File: {{cookiecutter.project_slug}}/manage.py [hunk 3]
```diff
@@ -5,6 +5,19 @@
 if __name__ == '__main__':
     os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'config.settings.local')
 
-    from django.core.management import execute_from_command_line
-
+    try:
```
Comment (pydanny) [inline {{cookiecutter.project_slug}}/manage.py]: I love this whole section!

File: {{cookiecutter.project_slug}}/config/settings/common.py (hunk 1 above)
Comment (luzfcb) [inline {{cookiecutter.project_slug}}/config/settings/common.py]: ok, I'll will remove and send in a separate pull request

**Comment by ssteinerx** [comment]
What is code coverage after this?  I know we had to kill the template coverage gizmo (though I think I figured out how to fix it).  

Do we have a test that asserts 100% code coverage on a newly minted project?

**Comment by ssteinerx** [comment]
@luzfcb Could you please merge master into this branch or rebase?  A bunch of my change on this branch are due to my branch being is up to date with master and it makes it hard to see what is different that is not due to that.

I have a couple of small changes (updating .gitignore, for example) that should go into the 1.10 release, but a pull request now would contain all the master branch updates too which is not what we want.

Thanks!

File: {{cookiecutter.project_slug}}/{{cookiecutter.project_slug}}/users/tests/test_admin.py [hunk 4]
```diff
@@ -27,8 +27,8 @@ def test_clean_username_false(self):
         # Instantiate the form with the same username as self.user
         form = MyUserCreationForm({
             'username': self.user.username,
-            'password1': '123456',
-            'password2': '123456',
+            'password1': 'notalamodespassword',
+            'password2': 'notalamodespassword',
```
Comment (jayfk) [inline {{cookiecutter.project_slug}}/{{cookiecutter.project_slug}}/users/tests/test_admin.py]: :shipit:

**Comment by pydanny** [comment]
@ssteinerx This isn't standard practice for projects, but we're not worrying about getting to 100% test coverage in a Django version update pull request. Here's why:

Getting to the 100% test coverage number in Cookiecutter Django is tedious work, especially when jumping versions. What we do is accept it, then as soon as it's done I jump to working on the code coverage. By making code coverage a separate effort, it allows for focus on getting the version update to just work. Some notes:

* If tests fail, we don't accept the version update pull request. Ever!
* Usually the incoming coverage score for tests on a version update is 95-100%.

**Comment by ssteinerx** [comment]
@pydanny -- I'm more concerned about testing our claim of "generating a Django project that has 100% test coverage out of the box."  IOW, not cookiecutter-django itself, but the project generated therefrom.

**Comment by pydanny** [comment]
@luzfcb :shipit:

**Comment by jayfk** [comment]
ship it

**Pull request merged**